scalafix-tests/output/src/main/scala/test/explicitResultTypes/ExplicitResultTypesPathDependent.scala

```diff
@@ -3,9 +3,14 @@ package test.explicitResultTypes
 object ExplicitResultTypesPathDependent {
   class Path {
     class B { class C }
-    implicit val x: _root_.test.explicitResultTypes.ExplicitResultTypesPathDependent.Path#B = new B
+    implicit val x: Path.this.B = new B
     implicit val y: x.C = new x.C
     def gimme(yy: x.C) = ???; gimme(y)
   }
   implicit val b: _root_.test.explicitResultTypes.ExplicitResultTypesPathDependent.Path#B = new Path().x
+  trait Foo[T] {
+    type Self
+    def bar: Self
+  }
+  implicit def foo[T]: _root_.test.explicitResultTypes.ExplicitResultTypesPathDependent.Foo[T]#Self = null.asInstanceOf[Foo[T]].bar
```

Comment:
Why it's not possible to handle this case?

Reply:
What do you mean? We correctly handle this case now. Previously we would discard the `[T]` parameter creating broken code.

This test puts only fully _root_ qualified names. The `Path.this` example above is as fully qualified as it can be, prefixing `Path` with `_root_...` will make the code not compile.

Reply:
Why it cannot be:

```scala
implicit def foo[T]: Foo[T]#Self = null.asInstanceOf[Foo[T]].bar
```

Reply:
Because the default setting fully qualifies all names with `_root_`. This is totally  unrelated to this PR.